### PR TITLE
Split function for maintenance exceptions

### PR DESCRIPTION
**Pull request by ziima**
Preparation for #23 - split function to determine whether the request is under maintenance checks.

### CODE REVIEW

File: maintenance_mode/middleware.py [hunk 1]
```diff
@@ -12,11 +8,8 @@
     from django.utils.deprecation import MiddlewareMixin
     __MaintenanceModeMiddlewareBaseClass = MiddlewareMixin
 
-from maintenance_mode import core, settings, utils
-from maintenance_mode.http import get_maintenance_response
-
-import re
-import sys
+from maintenance_mode import core
```
Comment (fabiocaccamo) [inline maintenance_mode/middleware.py]: `from maintenance_mode import core, http`

File: maintenance_mode/middleware.py [hunk 2]
```diff
@@ -12,11 +8,8 @@
...
+from maintenance_mode.http import get_maintenance_response, is_under_maintenance
```
Comment (fabiocaccamo) [inline maintenance_mode/middleware.py]: ~~`from maintenance_mode.http import get_maintenance_response, is_under_maintenance`~~

File: maintenance_mode/middleware.py [hunk 3]
```diff
@@ -26,78 +19,7 @@ def process_request(self, request):
...
-        if settings.MAINTENANCE_MODE_REDIRECT_URL:
-
-            redirect_url_re = re.compile(
-                settings.MAINTENANCE_MODE_REDIRECT_URL)
-
-            if redirect_url_re.match(request.path_info):
-                return None
-
-        return get_maintenance_response(request)
+        if is_under_maintenance(request):
```
Comment (fabiocaccamo) [inline maintenance_mode/middleware.py]: `if http.is_under_maintenance(request):`

File: maintenance_mode/middleware.py [hunk 4]
```diff
@@ -26,78 +19,7 @@ def process_request(self, request):
...
+            return get_maintenance_response(request)
```
Comment (fabiocaccamo) [inline maintenance_mode/middleware.py]: `return http.get_maintenance_response(request)`

File: maintenance_mode/http.py [hunk 5]
```diff
@@ -1,15 +1,83 @@
...
 import django
 from django.core.exceptions import ImproperlyConfigured
-from django.shortcuts import render, redirect
+from django.core.urlresolvers import NoReverseMatch, resolve, reverse
+from django.shortcuts import redirect, render
 from django.template import RequestContext
 from django.utils.cache import add_never_cache_headers
 from django.utils.module_loading import import_string
 
 from maintenance_mode import settings
```
Comment (fabiocaccamo) [inline maintenance_mode/http.py]: `from maintenance_mode import settings, utils`

File: maintenance_mode/http.py [hunk 6]
```diff
@@ -1,15 +1,83 @@
...
 
+from .utils import get_client_ip_address
```
Comment (fabiocaccamo) [inline maintenance_mode/http.py]: ~~`from .utils import get_client_ip_address`~~

File: maintenance_mode/http.py [hunk 7]
```diff
@@ -1,15 +1,83 @@
...
+        if settings.MAINTENANCE_MODE_GET_CLIENT_IP_ADDRESS:
+            try:
+                get_client_ip_address_func = import_string(settings.MAINTENANCE_MODE_GET_CLIENT_IP_ADDRESS)
+            except ImportError:
+                raise ImproperlyConfigured(
+                    'settings.MAINTENANCE_MODE_GET_CLIENT_IP_ADDRESS is not a valid function path.')
+            else:
+                client_ip_address = get_client_ip_address_func(request)
+        else:
+            client_ip_address = get_client_ip_address(request)
```
Comment (fabiocaccamo) [inline maintenance_mode/http.py]: `client_ip_address = utils.get_client_ip_address(request)`

File: maintenance_mode/http.py (hunk 5 above)
Comment (ziima) [reply maintenance_mode/http.py]: That's a very uncommon import style. Is there any reason for that?

Comment (fabiocaccamo) [reply maintenance_mode/http.py]: Why uncommon import style? We are using all the methods of the modules, so there is no difference.

Comment (ziima) [reply maintenance_mode/http.py]: I mean uncommon among python projects. Also imports from `maintenance_mode` often import whole modules, while the imports from Django import only functions or classes. I'd like to follow the style rules, but I don't see any.

Comment (ziima) [reply maintenance_mode/http.py]: Can we somehow sort this out as well?

**Pull request closed without merging**